CMakeLists.txt

```diff
@@ -8,15 +8,30 @@ set(USE_GPU OFF CACHE BOOL "Build Tensorflow with GPU support")
 set(CUDA_ROOT $ENV{CUDA_ROOT} CACHE PATH "CUDA root directory")
 set(CUDNN_ROOT $ENV{CUDNN_ROOT} CACHE PATH "CUDNN root directory")
 set(NCCL_ROOT $ENV{NCCL_ROOT} CACHE PATH "NCCL root directory")
+set(PYTHON_VERSION "2.7" CACHE STRING "Version of Python for bindings generation")
+set(NUM_MAKE_JOBS "" CACHE STRING "Number of jobs concurrently run during the build (passed as make -j parameter)")
+set(CUDA_VERSION $ENV{CUDA_VERSION} CACHE STRING "CUDA version")
+set(CUDNN_VERSION $ENV{CUDNN_VERSION} CACHE STRING "CUDNN version")
+
+if(${CMAKE_VERSION} VERSION_LESS "3.7.0") 
+       # ExternalProject_Add doesn't know the option SOURCE_SUBDIR in CMake before 3.7
+       file(DOWNLOAD "https://raw.githubusercontent.com/Kitware/CMake/v3.8.0/Modules/ExternalProject.cmake" ${CMAKE_CURRENT_SOURCE_DIR}/ExternalProject.cmake)
+       include(${CMAKE_CURRENT_SOURCE_DIR}/ExternalProject.cmake)
+else()
+  include(ExternalProject)
+endif()
 
-include(ExternalProject)
 file(MAKE_DIRECTORY ${CATKIN_DEVEL_PREFIX}/include)
 
+find_package(PythonLibs ${PYTHON_VERSION} REQUIRED)
+
 ExternalProject_Add(tensorflow_src
   GIT_REPOSITORY https://github.com/tensorflow/tensorflow.git
   GIT_TAG b874783ccdf4cc36cb3546e6b6a998cb8f3470bb
   PATCH_COMMAND bash -c "for i in ${CMAKE_SOURCE_DIR}/patches/*.patch\$<SEMICOLON> do git apply -p1 $i\$<SEMICOLON> done"
   UPDATE_COMMAND ""
+  BUILD_COMMAND make -j${NUM_MAKE_JOBS} install
```

Comment:
Should't `make` be called in the build step ?
```cmake
  BUILD_COMMAND make -j${NUM_MAKE_JOBS}
  INSTALL_COMMAND make install -j${NUM_MAKE_JOBS}
```

Reply:
I've found out that the install step of TF duplicates the work of the build step (because of the always invalidated `__force_rebuild` target), so it doesn't make sense to do the build step, then throw away almost all the work, and then do it again. It might deserve a comment on this line...